consul/src/test/scala/io/buoyant/consul/KvApiTest.scala

```diff
@@ -0,0 +1,225 @@
+package io.buoyant.consul.v1
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.Service
+import com.twitter.finagle.http.{Response, Request}
+import com.twitter.io.Buf
+import com.twitter.util.Future
+import io.buoyant.consul.{Forbidden, NotFound, UnexpectedResponse}
+import io.buoyant.test.Awaits
+import org.scalatest.FunSuite
+
+class KvApiTest extends FunSuite with Awaits {
+  val listBuf = Buf.Utf8("""["foo/bar/", "foo/baz/"]""")
+  val getBuf = Buf.Utf8("""foobar""")
+  val putOkBuf = Buf.Utf8("""true""")
+  val putFailBuf = Buf.Utf8("""false""")
+  var lastUri = ""
+
+  override val defaultWait = 2.seconds
+
+  def stubService(buf: Buf) = Service.mk[Request, Response] { req =>
+    val rsp = Response()
+    rsp.setContentTypeJson()
+    rsp.content = buf
+    rsp.headerMap.set("X-Consul-Index", "4")
+    lastUri = req.uri
+    Future.value(rsp)
+  }
+
+  test("list returns an indexed seq of key names") {
+    val service = stubService(listBuf)
+
+    val result = await(KvApi(service).list("foo/"))
+
+    assert(result.index == Some("4"))
+    assert(result.value.size == 2)
+    assert(result.value == List("foo/bar/", "foo/baz/"))
+  }
+
+  test("list reports wrong paths with an error") {
+    val failureService = Service.mk[Request, Response] { req =>
+      val rsp = Response()
+      rsp.headerMap.set("X-Consul-Index", "42")
+      rsp.setStatusCode(404)
+      lastUri = req.uri
+      Future.value(rsp)
+    }
+    try {
```

Comment:
this could be

`assertThrows[NotFound](await(...))`

(you have to import `io.buoyant.test.Exceptions`)